src/main/convert.js

```diff
@@ -107,17 +107,17 @@ export class ThriftFileConverter {
     `export type ${this.transformName(def.id.name)} = ${this.types.convert(def.valueType)};`;
 
   generateEnumMap = (def: Enum) => {
-    const header = `{`;
+    const header = '{';
     const values = def.definitions
       .map((d, index) => `  "${d.id.name}": ${d.value ? d.value.value : index},`)
       .join('\n');
-    const footer = `}`;
+    const footer = '}';
 
     return [header, values, footer].join('\n');
   };
 
   generateEnum = (def: Enum) => {
-    return `export const ${this.transformName(def.id.name)} = ${this.generateEnumMap(def)};`;
+    return `export type ${this.transformName(def.id.name)} = ${this.generateEnumMap(def)};`;
```

Comment:
Do we want the ability to grab enum values from code, or do we think those should be kept in sync separately? (Philosophical question)

Reply:
Based on what the generated code actually looks like, we really only use the keys of anything. For example:
```
/* thrift file */
struct ServiceError {
  1: optional bool ShouldRetry,
  2: optional string Message,
  3: optional map<string,string> Data,
  4: optional ErrorType Type,
}

/** There are bound to be additions to this list. ClientS should handle future additions by making
 string comparisons that won't break with the addition of new values */
enum ErrorType {
    INVALID_ARGUMENTS
    RATE_LIMIT_REACHED
    INTERNAL_ERROR
    PERMISSION_DENIED
    DB_ERROR
    DEPRECATED_ENDPOINT
}
```
```
/* js output */
export type ServiceErrorType = {|
  Message?: string,
  ShouldRetry?: boolean,
  Type?: $Keys<typeof ErrorTypeType>,
  Data?: {[string]: string},
|};

export const ErrorTypeType = {
  DB_ERROR: 4,
  DEPRECATED_ENDPOINT: 5,
  INTERNAL_ERROR: 2,
  INVALID_ARGUMENTS: 0,
  PERMISSION_DENIED: 3,
  RATE_LIMIT_REACHED: 1,
};
```

And if this is the case, `thriftrw-node` likely does the name to value translation on it's side.

That said, I know in our code particularly we do use these values during runtime to validate that the data types sent from the backend and the handlers for those datatypes in the frontend match up. We could always remove this code I guess but this specific pull request is most definitely not backwards compatible.

Reply:
I could add an export that has the values with a different name. Maybe:

```js
/* js output */
export type ServiceErrorType = {|
  Message?: string,
  ShouldRetry?: boolean,
  Type?: $Keys<typeof ErrorTypeType>,
  Data?: {[string]: string},
|};

export type ErrorTypeType = {
  DB_ERROR: 4,
  DEPRECATED_ENDPOINT: 5,
  INTERNAL_ERROR: 2,
  INVALID_ARGUMENTS: 0,
  PERMISSION_DENIED: 3,
  RATE_LIMIT_REACHED: 1,
};
export const ErrorType = {
  DB_ERROR: 4,
  DEPRECATED_ENDPOINT: 5,
  INTERNAL_ERROR: 2,
  INVALID_ARGUMENTS: 0,
  PERMISSION_DENIED: 3,
  RATE_LIMIT_REACHED: 1,
};
```